### DISCUSSION TITLE
Multi tenancy System.NullReferenceException: 'TenantId claim not found'  tenantClaim was null.

### DISCUSSION THREAD

**Discussion by mvmshakkir** [serenity-is/Serenity#7128] [Help]
my code is 
public static ClaimsPrincipal CreatePrincipal(IUserRetrieveService userRetriever, string username, string authType, int TenantId)
{
    if (userRetriever is null)
        throw new ArgumentNullException(nameof(userRetriever));

    if (username is null)
        throw new ArgumentNullException(nameof(username));

    var user = (UserDefinition)userRetriever.ByUsername(username);
    if (user == null)
        throw new ArgumentOutOfRangeException(nameof(username));

    if (authType == null)
        throw new ArgumentNullException(nameof(authType));

    var identity = new GenericIdentity(username, authType);
    identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, user.Id));
    identity.AddClaim(new Claim("TenantId", user.TenantId.ToInvariant())); // add tenant // Adding TenantId claim as string
    
        return new ClaimsPrincipal(identity);
} 

using System;
using System.Linq;
using System.Security.Claims;
namespace DailyStandUp
{
    public static class ClaimsPrincipalExtensions
    {
        public static int GetTenantId(this ClaimsPrincipal user)
        {
            if (user is null)
                throw new ArgumentNullException(nameof(user));

            var tenantClaim = user.Claims.FirstOrDefault(x => x.Type == "TenantId");
            if (tenantClaim is null)
                throw new NullReferenceException("TenantId claim not found");

            return int.Parse(tenantClaim.Value);
        }
    }
}
 here it throwing error System.NullReferenceException: 'TenantId claim not found'

tenantClaim was null.

**Comment by Zahar661** [comment]
Hi! Did you manage to solve it?
